dart/lib/src/sentry.dart

```diff
@@ -53,16 +57,20 @@ class Sentry {
   static Future<void> _initDefaultValues(
     SentryOptions options,
     AppRunner appRunner,
-    List<Integration> initialIntegrations,
+    List<Integration> initialIntegrations
   ) async {
-    // if no environment is set, we set 'production' by default, but if we know it's
-    // a non-release build, or the SENTRY_ENVIRONMENT is set, we read from it.
-    if (const bool.hasEnvironment('SENTRY_ENVIRONMENT') || !isReleaseMode) {
-      options.environment = const String.fromEnvironment(
-        'SENTRY_ENVIRONMENT',
-        defaultValue: 'debug',
-      );
-    }
+    // We infer the enviroment based on the release/non-release and profile
+    // constants.
+    var environment = options.platformChecker.isReleaseMode()
+        ? defaultEnvironment
+        : options.platformChecker.isProfileMode()
+            ? 'profile'
+            : 'debug';
+
+    // if the SENTRY_ENVIRONMENT is set, we read from it.
+    options.environment = const bool.hasEnvironment('SENTRY_ENVIRONMENT')
+        ? const String.fromEnvironment('SENTRY_ENVIRONMENT')
+        : environment;
```

Comment:
Does it throw if u read and there's nothing? Does it just return null instead?

```suggestion
    // if the SENTRY_ENVIRONMENT is set, we read from it.
    var environment = const String.fromEnvironment('SENTRY_ENVIRONMENT');
    if (environment == null) {
        options.platformChecker.isReleaseMode()
            ? defaultEnvironment
            : options.platformChecker.isProfileMode()
                ? 'profile'
                : 'debug';    
    }

```

Reply:
it reads a static field, so it won't throw.
returns `null` or the given `defaultValue`

Reply:
OK so the suggestion makes sense I'd say

Reply:
We check `bool.hasEnvironment('SENTRY_ENVIRONMENT')` beforehand, so it will not return null in the next line afterwards. I thing we are safe here, as we do this the same way some lines below: https://github.com/getsentry/sentry-dart/blob/b91b0c68f853505822e0b1e14a729fd100dde463/dart/lib/src/sentry.dart#L76

Reply:
I suggested we avoid this check because it's not helpful, we can just fetch the data and assing, and if it's null we run the rest of the code.